src/Servers/Kestrel/Transport.Quic/src/Internal/QuicStreamStack.cs

```diff
@@ -0,0 +1,144 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Transport.Quic.Internal
+{
+    // See https://github.com/dotnet/runtime/blob/master/src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/BufferSegmentStack.cs
+    internal struct QuicStreamStack
```

Comment:
Maybe we should reference this is equivalent to Http2Stack. Can we share this without killing perf? Might be worth adding to the CodeGenerator.

Reply:
Generics should be fine.